.github/workflows/e2e.yaml

```diff
@@ -143,79 +152,81 @@ jobs:
-    #     # CoreDNS should handle those domains and answer with NXDOMAIN instead of SERVFAIL
-    #     # otherwise pods stops trying to resolve the domain.
-    #     # Get the current config
-    #     original_coredns=$(/usr/local/bin/kubectl get -oyaml -n=kube-system configmap/coredns)
-    #     echo "Original CoreDNS config:"
-    #     echo "${original_coredns}"
-    #     # Patch it
-    #     fixed_coredns=$(
-    #       printf '%s' "${original_coredns}" | sed \
-    #         -e 's/^.*kubernetes cluster\.local/& net/' \
-    #         -e '/^.*upstream$/d' \
-    #         -e '/^.*fallthrough.*$/d' \
-    #         -e '/^.*forward . \/etc\/resolv.conf$/d' \
-    #         -e '/^.*loop$/d' \
-    #     )
-    #     echo "Patched CoreDNS config:"
-    #     echo "${fixed_coredns}"
-    #     printf '%s' "${fixed_coredns}" | /usr/local/bin/kubectl apply -f -
-
-    # - name: Run tests
-    #   run: |
-    #     export KUBERNETES_CONFORMANCE_TEST='y'
-    #     export E2E_REPORT_DIR=${PWD}/_artifacts
-
-    #     # Run tests
-    #     /usr/local/bin/ginkgo --nodes=25                \
-    #       --focus="\[Conformance\]"     \
-    #       --skip="Feature|Federation|machinery|PerformanceDNS|DualStack|Disruptive|Serial|Slow|KubeProxy|LoadBalancer|GCE|Netpol|NetworkPolicy|NodeConformance"   \
-    #       /usr/local/bin/e2e.test                       \
-    #       --                                            \
-    #       --kubeconfig=${PWD}/_artifacts/kubeconfig.conf     \
-    #       --provider=local                              \
-    #       --dump-logs-on-failure=false                  \
-    #       --report-dir=${E2E_REPORT_DIR}                \
-    #       --disable-log-dump=true
-
-    # - name: Upload Junit Reports
-    #   if: always()
-    #   uses: actions/upload-artifact@v2
-    #   with:
-    #     name: kind-junit-${{ env.JOB_NAME }}-${{ github.run_id }}
-    #     path: './_artifacts/*.xml'
-
-    # - name: Export logs
-    #   if: always()
-    #   run: |
-    #     /usr/local/bin/kind export logs --name ${KIND_CLUSTER_NAME} --loglevel=debug ./_artifacts/logs
-
-    # - name: Upload logs
-    #   if: always()
-    #   uses: actions/upload-artifact@v2
-    #   with:
-    #     name: kind-logs-${{ env.JOB_NAME }}-${{ github.run_id }}
-    #     path: ./_artifacts/logs
+    - name: Get Cluster status
+      run: |
+        # wait network is ready
+        sleep 5
+        /usr/local/bin/kubectl get nodes -o wide
+        /usr/local/bin/kubectl get pods -A
+        /usr/local/bin/kubectl wait --timeout=1m --for=condition=ready pods --namespace=kube-system -l k8s-app=kube-dns
+        # smoke test
+        /usr/local/bin/kubectl run test --image httpd:2
+        /usr/local/bin/kubectl wait --timeout=1m --for=condition=ready pods test
+        /usr/local/bin/kubectl get pods -A -owide
+        /usr/local/bin/kubectl wait --timeout=1m --for=condition=ready pods --namespace=kube-system -l k8s-app=konnectivity-agent
```

Comment:
It may be worth also waiting for `-l k8s-app=konnectivity-server` to be ready.

(In this 1 control-plane node cluster is is nearly the same, but in a multi control-plane cluster it would be needed.)

Reply:
I'm assuming you're talking about waiting the pod of ANP server with 
readiness probe.(now, the pod of ANP server have not readiness probe)

konnectivity-server'readiness  indicates that at least one Konnectivity Agent is connected.

konnectivity-agent'readiness indicates  that the client is connected to at least one proxy server.

Therefore, if readiness is added at the same time, both will enter an infinite loop.

The reason is ANP agent is using kubernetes svc to connect ANP server and ANP server is running with daemonset.

Remind me if i missed something,Thanks.

Reply:
> I'm assuming you're talking about waiting the pod of ANP server with readiness probe.(now, the pod of ANP server have not readiness probe)
> 
> konnectivity-server'readiness indicates that at least one Konnectivity Agent is connected.
> 
> konnectivity-agent'readiness indicates that the client is connected to at least one proxy server.

Yes to above.

> Therefore, if readiness is added at the same time, both will enter an infinite loop.

I don't expect this, because waiting on agent (or any) readiness does not depend on control plane egress.

> 
> The reason is ANP agent is using kubernetes svc to connect ANP server and ANP server is running with daemonset.
> 
> Remind me if i missed something,Thanks.

What really matters (for a given proxy request ,like `kubectl logs`) is whether the apiserver that handles the request has a konnectivity-server with at least one useful agent. Since the agent readiness only covers "at least 1", it is insufficient in this case. (There was discussion on that feature to add agent readiness mode "connected to all servers" but it is not implemented.) This would be the gap, fixed by waiting for all konnectivity-server to be ready.